controllers/cluster_controller_test.go

```diff
@@ -1498,6 +1498,60 @@ func TestClusterReconcilerValidateManagementEksaVersionFail(t *testing.T) {
 	g.Expect(err).To(HaveOccurred())
 }
 
+func TestClusterReconcilerNotAvailableEksaVersion(t *testing.T) {
+	version := test.DevEksaVersion()
+	config, _ := baseTestVsphereCluster()
+	config.Cluster.Name = "test-cluster"
+	config.Cluster.Spec.ManagementCluster = anywherev1.ManagementCluster{Name: "management-cluster"}
+	config.Cluster.Spec.BundlesRef = nil
+	config.Cluster.Spec.EksaVersion = &version
+
+	mgmt := config.DeepCopy()
+	mgmt.Cluster.Name = "management-cluster"
+	mgmt.Cluster.Spec.BundlesRef = nil
+	mgmt.Cluster.Spec.EksaVersion = &version
+
+	g := NewWithT(t)
+
+	objs := make([]runtime.Object, 0, 4+len(config.ChildObjects()))
+	objs = append(objs, config.Cluster, mgmt.Cluster)
+
+	for _, o := range config.ChildObjects() {
+		objs = append(objs, o)
+	}
+
+	testClient := fake.NewClientBuilder().WithRuntimeObjects(objs...).Build()
+
+	mockCtrl := gomock.NewController(t)
+	providerReconciler := mocks.NewMockProviderClusterReconciler(mockCtrl)
+	iam := mocks.NewMockAWSIamConfigReconciler(mockCtrl)
+	clusterValidator := mocks.NewMockClusterValidator(mockCtrl)
+	registry := newRegistryMock(providerReconciler)
+	mockPkgs := mocks.NewMockPackagesClient(mockCtrl)
+
+	ctx := context.Background()
+	log := testr.New(t)
+	logCtx := ctrl.LoggerInto(ctx, log)
+
+	iam.EXPECT().EnsureCASecret(logCtx, gomock.AssignableToTypeOf(logr.Logger{}), sameName(config.Cluster)).Return(controller.Result{}, nil)
+	clusterValidator.EXPECT().ValidateManagementClusterName(logCtx, gomock.AssignableToTypeOf(logr.Logger{}), sameName(config.Cluster)).Return(nil)
+
+	r := controllers.NewClusterReconciler(testClient, registry, iam, clusterValidator, mockPkgs, nil)
+
+	req := clusterRequest(config.Cluster)
+	_, err := r.Reconcile(logCtx, req)
+
+	g.Expect(err).To(HaveOccurred())
+	eksaCluster := &anywherev1.Cluster{}
+
+	err = testClient.Get(context.TODO(), req.NamespacedName, eksaCluster)
+	if apierrors.IsNotFound(err) {
+		t.Fatal("expected to find cluster")
+	}
+	g.Expect(eksaCluster.Status.FailureReason).ToNot(BeNil())
+	g.Expect(string(*eksaCluster.Status.FailureReason)).To(ContainSubstring(string(anywherev1.EksaVersionInvalidReason)))
```

Comment:
nit: I think gomega has a `HaveValue` or something like that that allows you to assert pointer values. So you can avoid the not nil assert and the casting

Also, can we assert the error message as well?